src/stdlib/test/double-conversion/test-fast-dtoa.cc

```diff
@@ -0,0 +1,395 @@
+            CHECK_EQ(current_test.representation, buffer.start());
+        }
+        CHECK(succeeded*1.0/total > 0.99);
+        CHECK(needed_max_length);
+}
+
+
+TEST(FastDtoaGayShortestSingle) {
+        char buffer_container[kBufferSize];
+        Vector<char> buffer(buffer_container, kBufferSize);
+        bool status;
+        int length;
+        int point;
+        int succeeded = 0;
+        int total = 0;
+        bool needed_max_length = false;
+
+        Vector<const PrecomputedShortestSingle> precomputed =
+        PrecomputedShortestSingleRepresentations();
+        for (int i = 0; i < precomputed.length(); ++i) {
+            const PrecomputedShortestSingle current_test = precomputed[i];
+            total++;
+            float v = current_test.v;
+            status = FastDtoa(v, FAST_DTOA_SHORTEST_SINGLE, 0, buffer, &length, &point);
+            CHECK(kFastDtoaMaximalSingleLength >= length);
+            if (!status) continue;
+            if (length == kFastDtoaMaximalSingleLength) needed_max_length = true;
+            succeeded++;
+            CHECK_EQ(current_test.decimal_point, point);
+            CHECK_EQ(current_test.representation, buffer.start());
+        }
+        CHECK(succeeded*1.0/total > 0.98);
+        CHECK(needed_max_length);
+}
+
+
+TEST(FastDtoaGayPrecision) {
+        char buffer_container[kBufferSize];
+        Vector<char> buffer(buffer_container, kBufferSize);
+        bool status;
+        int length;
+        int point;
+        int succeeded = 0;
+        int total = 0;
+        // Count separately for entries with less than 15 requested digits.
+        int succeeded_15 = 0;
+        int total_15 = 0;
+
+        Vector<const PrecomputedPrecision> precomputed =
+        PrecomputedPrecisionRepresentations();
+        for (int i = 0; i < precomputed.length(); ++i) {
+            const PrecomputedPrecision current_test = precomputed[i];
+            double v = current_test.v;
+            int number_digits = current_test.number_digits;
+            total++;
+            if (number_digits <= 15) total_15++;
+            status = FastDtoa(v, FAST_DTOA_PRECISION, number_digits,
+                              buffer, &length, &point);
+            CHECK(number_digits >= length);
+            if (!status) continue;
+            succeeded++;
+            if (number_digits <= 15) succeeded_15++;
+            TrimRepresentation(buffer);
+            CHECK_EQ(current_test.decimal_point, point);
+            CHECK_EQ(current_test.representation, buffer.start());
+        }
+        // The precomputed numbers contain many entries with many requested
+        // digits. These have a high failure rate and we therefore expect a lower
+        // success rate than for the shortest representation.
+        CHECK(succeeded*1.0/total > 0.85);
+        // However with less than 15 digits almost the algorithm should almost always
+        // succeed.
+        CHECK(succeeded_15*1.0/total_15 > 0.9999);
+}
```

Comment:
A lot of the `double-conversion` code, if not all, should be able to be dropped. We ended up not using it in production (going with https://github.com/fmtlib/fmt instead). The tests are still present though, so we'll need to revisit them a bit.

Reply:
This has been mostly done. I'll resolve this now and write a followup comment on one of the remaining parts.